api/bootstrap/v1beta1/k0s_types.go

```diff
@@ -66,7 +66,7 @@ type K0sWorkerConfigSpec struct {
 
 	// Files specifies extra files to be passed to user_data upon creation.
 	// +kubebuilder:validation:Optional
-	Files []cloudinit.File `json:"files,omitempty"`
+	Files []File `json:"files,omitempty"`
```

Comment:
Could we not enhance the `cloudinit.File` struct directly instead of creating new one? I see that we "inherit" from it but somehow feels bit wonky 😄

Reply:
Well, technically we can, but we will have to move ContentSource and ContentSourceRef structs to the cloudinit package. Is it ok?

Reply:
IMO it makes sense to have the cloudinit related things in same package

Reply:
Well, not sure it's directly related to cloud-init, it's more a type that will be converted to cloudinit.File. 

I tried to simply move it to the cloudinit package, but it breaks `zz_generated.deepcopy.go` generation, because it requires `DeepCopyInto` method of the complex struct (with pointers to other structs):

```
if in.Files != nil {
	in, out := &in.Files, &out.Files
	*out = make([]cloudinit.File, len(*in))
	for i := range *in {
		(*in)[i].DeepCopyInto(&(*out)[i])
	}	
}
```

Reply:
> I tried to simply move it to the cloudinit package, but it breaks zz_generated.deepcopy.go generation, because it requires DeepCopyInto method of the complex struct (with pointers to other structs)

ok, that's enough proof that it's more effort to move it rather than have it as is. so IMO ok to go ahead as-is!